diff --git a/src/uvc-v4l2.cpp b/src/uvc-v4l2.cpp
index e97dfda..13f0596 100644
--- a/src/uvc-v4l2.cpp
+++ b/src/uvc-v4l2.cpp
@@ -726,24 +726,14 @@ namespace rsimpl
         std::vector<std::shared_ptr<device>> query_devices(std::shared_ptr<context> context)
         {
             // Check if the uvcvideo kernel module is loaded
-            std::ifstream modules("/proc/modules");
-            std::string modulesline;
-            std::regex regex("uvcvideo.* - Live.*");
-            std::smatch match;
-            bool module_found = false;
-
-
-            while(std::getline(modules,modulesline) && !module_found)
-            {
-                module_found = std::regex_match(modulesline, match, regex);
-            }
-
-            if(!module_found)
+            struct stat info;
+            int err = stat("/sys/module/uvcvideo/", &info);
+            if (err == -1  // errno==ENOENT, typically
+  		|| !S_ISDIR(info.st_mode) )
             {
                 throw std::runtime_error("uvcvideo kernel module is not loaded");
             }
 
-
             // Enumerate all subdevices present on the system
             std::vector<std::unique_ptr<subdevice>> subdevices;
             DIR * dir = opendir("/sys/class/video4linux");
